pkg/apis/maistra/conversion/extension_providers.go

```diff
@@ -18,34 +19,153 @@ func populateExtensionProvidersValues(in *v2.ControlPlaneSpec, values map[string
 		}
 	}
 
+	var extensionProvidersValues []map[string]interface{}
 	for _, ext := range in.ExtensionProviders {
-		if ext.Prometheus == nil {
-			return fmt.Errorf("extension provider entry %s does not define any provider - it must specify one of: prometheus", ext.Name)
+		if ext.Prometheus == nil && ext.EnvoyExtAuthzHTTP == nil {
+			return fmt.Errorf("extension provider %s does not define any provider - it must specify one of: prometheus or envoyExtAuthzHttp", ext.Name)
 		}
-		prometheus := []map[string]interface{}{
-			{
+		if ext.Prometheus != nil && ext.EnvoyExtAuthzHTTP != nil {
+			return fmt.Errorf("extension provider %s must specify only one type of provider: prometheus or envoyExtAuthzHttp", ext.Name)
+		}
+		if ext.Prometheus != nil {
+			extensionProvidersValues = append(extensionProvidersValues, map[string]interface{}{
+				// TODO: check empty string
 				"name":       ext.Name,
```

Comment:
Why do we call this `name` instead of `type`?

In Kubernetes, you usually have a `type` field and then another field to configure the thing (the name of this field matches the value of the `type`, but the first letter is lower-cased).

Reply:
> Why do we call this name instead of type?

Because this 1:1 copy of the upstream API.